adapters/oidc/spring-security/src/main/java/org/keycloak/adapters/springsecurity/filter/KeycloakPreAuthActionsFilter.java

```diff
@@ -78,7 +79,16 @@ public void doFilter(ServletRequest request, ServletResponse response, FilterCha
             throws IOException, ServletException {
 
         HttpFacade facade = new SimpleHttpFacade((HttpServletRequest)request, (HttpServletResponse)response);
-        nodesRegistrationManagement.tryRegister(deploymentContext.resolveDeployment(facade));
+        KeycloakDeployment deployment = deploymentContext.resolveDeployment(facade);
+        
+        if (deployment == null) {
+            return;
+        }
+
+        if (deployment.isConfigured()) {
```

Comment:
This will stop other things from being executed won't it? So shouldn't it just be "if (deployment != null && deployment.isConfigured()" instead of the check above that returns without continuing the chain.

Reply:
Other filters in the chain always expect a deployment instance. The chain continues as long as you have a deployment 9configured or not), which is the expected behavior.